modules/core/src/ocl.cpp

```diff
@@ -2977,8 +2977,11 @@ int Kernel::set(int i, const KernelArg& arg)
     cl_int status = 0;
     if( arg.m )
     {
-        int accessFlags = ((arg.flags & KernelArg::READ_ONLY) ? ACCESS_READ : 0) +
-                          ((arg.flags & KernelArg::WRITE_ONLY) ? ACCESS_WRITE : 0);
+        AccessFlag accessFlags = ACCESS_READ | ACCESS_WRITE;
+        if (!(arg.flags & KernelArg::READ_ONLY))
+            accessFlags &= ~ACCESS_READ;
+        if (!(arg.flags & KernelArg::WRITE_ONLY))
+            accessFlags &= ~ACCESS_WRITE;
         bool ptronly = (arg.flags & KernelArg::PTR_ONLY) != 0;
```

Comment:
@alalek @hrnr We do not have `ACCESS_NONE` to write something like `((arg.flags & KernelArg::READ_ONLY) ? ACCESS_READ : ACCESS_NONE) | ((arg.flags & KernelArg::WRITE_ONLY) ? ACCESS_WRITE : ACCESS_NONE)` and even the `ACCESS_MASK` is not covering the `FAST_ACCESS` bit...
In this case, all I can think of is the negative logic. Is there any better idea to do this more efficiently?

Reply:
For any "flags" enum, "0" is default "none" value.

I would like to avoid introducing separate meaningless enum value like `ACCESS_NONE`. Perhaps we can write here `(AccessFlag)0` instead.

It is better to drop "negative logic" in code, like your suggestion.

Alternatively we can try to overload `operator - ()` for "flag" enums (but we should make it safe).